zip-proportional-output-fee-mechanism-pofm.rst

```diff
@@ -0,0 +1,225 @@
+::
+
+  ZIP: 317
+  Title: Proportional Output Fee Mechanism (POFM)
+  Owners: Aditya Bharadwaj <nighthawk24@gmail.com>
+  Credits: Madars Virza
+           Kris Nuttycombe
+  Status: Draft
+  Category: Wallet
+  Created: 2022-08-15
+  License: MIT
+  Discussions-To: <https://forum.zcashcommunity.com/t/zip-proportional-output-fee-mechanism-pofm/42808>
+  Pull-Request: <https://github.com/zcash/zips/pull/631>
+
+
+Terminology
+===========
+
+The key words "MUST", "SHOULD", "SHOULD NOT", "MAY", "RECOMMENDED",
+"OPTIONAL", and "REQUIRED" in this document are to be interpreted as
+described in RFC 2119. [#RFC2119]_
+
+"Z" refers to shielded address.
+"T" refers to transparent address.
+
+"POFM" refers to the proposed change in conventional fee as described in 
+this ZIP.
+
+The term "default transaction fee" in this document is in reference
+to Z to Z, T to Z & Z to T transaction fee paid to miners on the Zcash network
+for the work on including the shielded transaction in a block.
+
+The terms "Mainnet, "Testnet", and "zatoshi" in this document are defined as in [#protocol-networks]_.
+
+
+Abstract
+========
+
+The goal of this ZIP is to change the conventional transaction fees for
+Shielded transactions and get buy-in from wallet developers, miners & Zcash users.
+
+With an updated transaction fee formula, miners will be compensated fairly for
+including transactions with a high number of inputs or outputs, while still allowing
+low fees for regular shielded transaction use cases.
+
+Once the new fee policy is enforced, this will additionally increase the cost of creating
+transactions that collectively have many inputs or outputs, discouraging usage patterns
+that cause either intentional or unintentional denial of service.
+
+
+Motivation
+============
+
+In light of recent network activity, it is time to review and update the 
+standard 1,000 zatoshi transaction fees set in ZIP 313 [#zip-0313]_.
+
+
+The conventional transaction fee presently is 0.00001 ZEC or 1,000 zats per
+ZIP-313, that allowed exploration of novel use cases of the Zcash blockchain.
+The Zcash network has operated for almost 2 years of 1,000 zats fee per shielded 
+transaction without consideration for the total number of inputs and outputs in the transaction.
+This has resulted in high output transactions with 1,100 outputs costing the same as 
+transactions with 2 outputs.
+
+
+Requirements for gathering consensus
+-------------------------
+
+Wallet developers SHOULD update the fees to the proposed formula by Madars and 
+Kris Nuttycombe [#madars-1]_
+
+min_fee = base_fee * max(1, #inputs + #outputs - 4)
```

Comment:
There is an open poll to gauge sentiment for a `base_fee` of 0.0001 ZEC for Transparent transactions, and 0.00001 ZEC for Shielded, Shielding, and De-shielding transactions.

https://twitter.com/aiyadt/status/1559621424033169409

Okay, let's change the contact to 3.

Reply:
Is there any merit to the stray thought that inputs are perhaps more favored than outputs?

In my case, i may have a lot of small txs coming in and then I z_mergetoaddress to send them all to a colder wallet. Would z_mergetoaddress be many inputs and 1 output? I haven't looked under the covers but this would be my intuition. Would this proposal, not differentiating between inputs and outputs, essentially make notes smaller than the base fee unusable in practice?

Sweeping up dust into a single output should perhaps be cheaper than spraying dust into many outputs?

Are inputs intrinsically more "virtuous" than outputs?

Reply:
For Orchard, there is no difference between inputs and outputs. A 1-in 100-out Orchard tx looks the same as a 100-in 1-out Orchard tx.

Reply:
For Sapling, inputs and outputs are costly in different ways (mainly, inputs require verification of spend authorization signatures, while outputs require trial decryption). Neither is inherently more costly than the other.